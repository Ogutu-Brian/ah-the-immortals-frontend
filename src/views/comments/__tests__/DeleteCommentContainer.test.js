import React from 'react';
import { Router } from 'react-router-dom';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { createBrowserHistory } from 'history';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import toJson from 'enzyme-to-json';
import { Button } from 'react-bootstrap';
import DeleteComment from '../DeleteCommentContainer';

describe('Tests DeleteCommentContainer', () => {
  const store = configureStore([thunk])({
    comments: { data: {comments: []}, error: {}}
  });
  const props = {
    deleteCommentAction: jest.fn(),
    id: 3,
    slug: 'string'
  };
  const history = createBrowserHistory();
  const wrapper = mount (
    <Router history={history}>
      <Provider store={store}>
        <DeleteComment {...props} />
      </Provider>
    </Router>
  );
  it('tests mounting of container', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('tests click of delete button', () => {
    const btn = wrapper.find(Button);
    btn.simulate('click', {
      preventDefault: jest.fn(),
    });
    expect(btn.length).toEqual(1);
  });
});
