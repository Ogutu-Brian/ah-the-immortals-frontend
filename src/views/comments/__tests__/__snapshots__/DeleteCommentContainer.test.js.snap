// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests DeleteCommentContainer tests mounting of container 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Connect(DeleteComment)
      deleteCommentAction={[MockFunction]}
      id={3}
      slug="string"
    >
      <DeleteComment
        data={
          Object {
            "comments": Array [],
          }
        }
        deleteCommentAction={[Function]}
        id={3}
        slug="string"
      >
        <Button
          active={false}
          className="card-link btn-comment"
          disabled={false}
          onClick={[Function]}
          type="button"
          variant="danger"
        >
          <button
            className="card-link btn-comment btn btn-danger"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <i
              className="fas fa-trash-alt"
            />
          </button>
        </Button>
      </DeleteComment>
    </Connect(DeleteComment)>
  </Provider>
</Router>
`;
